tjreports/administrator/controllers/tjreports.php

```diff
@@ -55,4 +55,68 @@ public function setRedirect($url, $msg = null,$type = null)
 
 		parent::setRedirect($url, $msg, $type);
 	}
+
+	/**
+	 * Discover installed plugins
+	 *
+	 * @return  JModel
+	 *
+	 * @since   1.6
+	 */
+	public function discover()
+	{
+		$db    = JFactory::getDbo();
+		$query = $db->getQuery(true);
+		$query->select('element');
+		$query->from($db->quoteName('#__extensions'));
+		$query->where($db->quoteName('type') . "=" . $db->quote('plugin'));
+		$query->where($db->quoteName('folder') . "=" . $db->quote('tjreports'));
+		$db->setQuery($query);
+		$allPlugins = $db->loadObjectList();
+
+		$query = $db->getQuery(true);
+		$query->select('plugin', 'element');
+		$query->from($db->quoteName('#__tj_reports'));
+		$db->setQuery($query);
+		$tjReportsPlugins = $db->loadObjectList();
+
+		$json  = json_encode($allPlugins);
```

Comment:
Why is all this JSON needed ? You can simplify this by

1. Fetch all the plugins
2. Fetch all unique plugins for which there is a row in tj reports table
3. Loop through all items in step 1, and discover the ones which are not in step 2 results.